HelpSource/Overviews/MidiPatterns.schelp

```diff
@@ -0,0 +1,184 @@
+title:: How to control midi events using patterns
+categories:: Midi, Streams-Events-Patterns
+summary:: An overview of midi commands and how to use them with Pbind, etc.
+
+This document is a brief overview of some commonly used midi commands
+available in the midievent part of LINK::Classes/Event:: and which arguments
+they offer.
+
+To use them in a event pattern such as LINK::Classes/Pbind::, you need to first
+initialise a midi out device.
+
+CODE::
+
+// This posts a list of available midi devices
+// Note their names and bus names and use them in the next step
+MIDIClient.init;
+
+// Initialise a midi out using the name and bus name (see above)
+// As an example:
+m = MIDIOut.newByName("IAC Driver", "Bus 1");
+::
+
+In the examples below, it is assumed that you have initialised a midi out.
+
+NOTE::All midi numbers in SuperCollider are 0-indexed. That is, what on most
+devices is called "midi channel 1" is actually midi channel 0 in SuperCollider
+and vice-versa.::
+
+Now, to use it in a Pbind:
+
+CODE::
+p = Pbind(
+	// These keys are common to all midi commands
+    \type, \midi,
+    \midiout, m,
+    \midicmd, \noteOn, // Choose which command to use
+
+    \chan, 0,
+	// degree is converted to midinote, not just frequency
+    \degree, Pwhite(-7, 12, inf),
+    \dur, Pwrand([0.25, Pn(0.125, 2)], #[0.8, 0.2], inf),
+    \legato, sin(Ptime(inf) * 0.5).linexp(-1, 1, 1/3, 3),
+    \amp, Pexprand(0.5, 1.0, inf)
+).play(quant: 1);
+)
+
+p.stop;
+::
+
+SECTION:: \noteOn and \noteOff
+
+The most common midi commands are \noteOn and \noteOff. They are used to play
+notes. \noteOff is automatically transmitted after the note duration when you
+use the \noteOn command.
+
+CODE::
+p = Pbind(
+	// These keys are common to all midi commands
+    \type, \midi,
+    \midiout, m,
+    \midicmd, \noteOn, // Choose which command to use
+
+    \chan, 0,
+	// degree is converted to midinote, not just frequency
+    \degree, Pwhite(-7, 12, inf),
+    \dur, Pseq([1,2,0.5], inf),
+	// amp is converted to velocity
+    \amp, Pexprand(0.5, 1.0, inf)
+).play(quant: 1);
+)
+
+p.stop;
+::
+
+SECTION:: control change
+
+Send control change messages.
+
+CODE::
+p = Pbind(
+	// These keys are common to all midi commands
+    \type, \midi,
+    \midiout, m,
+    \midicmd, \control,
+
+    \chan, 0,
+	\ctlNum, 0, // cc number
+	\control, Pwhite(0, 127, inf), // cc value
+    \dur, Pseq([1,2,0.5], inf),
+
+).play(quant: 1);
+)
+
+p.stop;
+::
+
+SECTION:: \polyTouch
+
+CODE::
+p = Pbind(
+	// These keys are common to all midi commands
```

Comment:
maybe instead of `// These keys are common to all midi commands` it should be
` //  the \type, \midiout, and \midicmd keys are common to all midi commands`

I stared at these examples for a minute before understanding (though I might be dense!)

Either way this is super helpful...